src/parser_yin.c

```diff
@@ -3318,19 +3294,15 @@ yin_parse_element_generic(struct yin_parser_ctx *ctx, const char *name, size_t n
         LY_CHECK_ERR_GOTO(!last->stmt, LOGMEM(ctx->xml_ctx.ctx); ret = LY_EMEM, err);
         /* attributes with prefix are ignored */
         if (!temp_prefix) {
-            if (dynamic) {
-                last->arg = lydict_insert_zc(ctx->xml_ctx.ctx, out);
-                LY_CHECK_ERR_GOTO(!last->arg, free(out); LOGMEM(ctx->xml_ctx.ctx); ret = LY_EMEM, err);
-            } else {
-                last->arg = lydict_insert(ctx->xml_ctx.ctx, out, out_len);
-                LY_CHECK_ERR_GOTO(!last->arg, LOGMEM(ctx->xml_ctx.ctx); ret = LY_EMEM, err);
-            }
+            INSERT_STRING(ctx->xml_ctx.ctx, last->arg, dynamic, out, out_len);
+            LY_CHECK_ERR_GOTO(!last->arg, ret = LY_EMEM, err);
         }
     }
 
     /* parse content of element */
     ret = lyxml_get_string(&ctx->xml_ctx, data, &out, &out_len, &out, &out_len, &dynamic);
     if (ret == LY_EINVAL) {
+        LY_CHECK_GOTO(ret, err);
```

Comment:
Since `ret` is always `LY_EINVAL`, the rest of the block after `LY_CHECK_GOTO` seems as a dead code.